eclair-core/src/main/scala/fr/acinq/eclair/io/OpenChannelInterceptor.scala

```diff
@@ -0,0 +1,293 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.io
+
+import akka.actor.Status
+import akka.actor.typed.scaladsl.adapter.TypedActorRefOps
+import akka.actor.typed.scaladsl.{ActorContext, Behaviors}
+import akka.actor.typed.{ActorRef, Behavior}
+import fr.acinq.bitcoin.scalacompat.Crypto.PublicKey
+import fr.acinq.bitcoin.scalacompat.{BtcDouble, ByteVector32, Satoshi, Script}
+import fr.acinq.eclair.Features.Wumbo
+import fr.acinq.eclair.blockchain.OnchainPubkeyCache
+import fr.acinq.eclair.channel._
+import fr.acinq.eclair.channel.fsm.Channel
+import fr.acinq.eclair.io.Peer.SpawnChannelNonInitiator
+import fr.acinq.eclair.io.PendingChannelsRateLimiter.AddOrRejectChannel
+import fr.acinq.eclair.wire.protocol
+import fr.acinq.eclair.wire.protocol.Error
+import fr.acinq.eclair.{AcceptOpenChannel, CltvExpiryDelta, Features, InitFeature, InterceptOpenChannelPlugin, InterceptOpenChannelReceived, InterceptOpenChannelResponse, MilliSatoshi, NodeParams, RejectOpenChannel, ToMilliSatoshiConversion}
+import scodec.bits.ByteVector
+
+import scala.concurrent.duration.{DurationInt, FiniteDuration}
+import scala.reflect.ClassTag
+
+/**
+ * Child actor of a Peer that handles accepting or rejecting a channel open request initiated by a remote peer and
+ * configuring local parameters for all open channel requests. It only handles one channel request at a time.
+ * If a concurrent request comes while still evaluating a previous one, the later request is immediately rejected.
+ *
+ * This simplifies the state machine and ensures that we do not introduce subtle race conditions when checking limits,
+ * since some of the conditions will be asynchronous (e.g. asking the router whether the node is public or private).
+ *
+ * When initiator (peer.OpenChannel):
+ *  - verify synchronous conditions against eclair.conf
+ *  - create LocalParams from nodeParams
+ *  - send SpawnChannelInitiator to parent Peer
+ *
+ * When non-initiator (protocol.OpenChannel, protocol.OpenDualFundedChannel):
+ *  - verify synchronous conditions against eclair.conf
+ *  - if the peer is not whitelisted to ignore rate-limits:
+ *    - ask the PendingChannelsRateLimiter if we should allow the channel
+ *    - if the rate limit is reached, reply to parent Peer with `OutgoingMessage(Error)`
+ *  - create LocalParams from nodeParams
+ *  - if a channel interceptor plugin is registered, ask it whether to allow the channel request:
+ *    - if the plugin rejects the request, reply to parent Peer with `OutgoingMessage(Error)`
+ *    - if the plugin doesn't respond in time, reply to parent Peer with `OutgoingMessage(Error)`
+ *    - if the plugin allows the request, update LocalParams with the plugin's response
+ *  - accept the channel, reply to parent Peer with `SpawnChannelNonInitiator`
```

Comment:
I don't think it's a good idea to have such a detailed comment of the flow so far from the actual code. We will eventually update the code and will forget to update that comment, which will just become misleading. I'd rather remove that part and add comments on each behavior function to explain what it does when it's not obvious from the code itself.